### PR TITLE
Add basic support for insert/update/delete/subquery in plan2

### PR DESCRIPTION
**Pull request by ouyuanning**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [x] Code Refactoring

**Which issue(s) this PR fixes:**

issue #

**What this PR does / why we need it:**

add basic support for insert/update/delete/subquery 
many corners are not covered yet. 
basic test case will be found in /pkg/plan2/build_test.go

**Special notes for your reviewer:**

type checking and type conversion are a little complicated, I need more time to work for that.

happy holidays.  :)

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/sql/plan2/build_function.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+	"fmt"
+
+	"github.com/matrixorigin/matrixone/pkg/errno"
+	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"github.com/matrixorigin/matrixone/pkg/sql/errors"
+	"github.com/matrixorigin/matrixone/pkg/sql/parsers/tree"
+)
+
+func checkFloatType(typ plan.Type_TypeId, alias string) error {
+	switch typ {
```
Comment (fengttt) [inline pkg/sql/plan2/build_function.go:27]: why not use switch typ { 
     case a, b, c:
}

File: pkg/sql/plan2/build_function.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+		default:
+			if argType != args[idx].Typ.Id {
+				return errors.New(errno.SyntaxErrororAccessRuleViolation, fmt.Sprintf("type error: arg '%v' is not %v", args[idx].Alias, plan.Type_TypeId_name[int32(argType)]))
+			}
+		}
+	}
+	return nil
+}
+
+func coverArgToHigh(args []*plan.Expr) error {
```
Comment (fengttt) [inline pkg/sql/plan2/build_function.go:133]: I think you need to add a cast function -- basically, 
int32 + int64      ->    translate into    int64(int32) + int64

Comment (fengttt) [reply pkg/sql/plan2/build_function.go:133]: It is probably better to have a implicit cast table, saying type x can be cast to type y implicitly.   Then we look at this table.

For example, we know int8 -> int16 is ok, uint8 -> uint16 is ok, uint8 -> int16 is also ok. etc etc.

File: pkg/sql/plan2/build_insert.go [hunk 3]
```diff
@@ -14,8 +14,197 @@
 
 package plan2
 
-import "github.com/matrixorigin/matrixone/pkg/sql/parsers/tree"
```
Comment (fengttt) [inline pkg/sql/plan2/build_insert.go:17]: How does insert into dest select * from src work?   
I think insert should be just algebraized as,
insert -> insert node, which has a input (that is, child)
select or values -> a query tree

File: pkg/sql/plan2/build_subquery.go [hunk 4]
```diff
@@ -23,6 +23,65 @@ import (
...
+	//把子查询的节点放到前面去
+	query.Steps = []int32{int32(nowLength - 1)}
+	query.Nodes = append(query.Nodes[nowLength:], query.Nodes[:nowLength]...)
+
+	return &plan.Expr{
+		Expr: &plan.Expr_Sub{
+			Sub: expr,
+		},
+		Typ: &plan.Type{
+			Id:       plan.Type_ARRAY,
```
Comment (fengttt) [inline pkg/sql/plan2/build_subquery.go:83]: Why it is an ARRAY?

File: pkg/sql/plan2/build_function.go (hunk 1 above)
Comment (ouyuanning) [reply pkg/sql/plan2/build_function.go:27]: yes, better one.  I have change the code in the last commit

File: pkg/sql/plan2/build_function.go (hunk 2 above)
Comment (ouyuanning) [reply pkg/sql/plan2/build_function.go:104]: type check and type cast will be done in next commit/PR，I need more research and design work。
And I got the  suggestion

File: pkg/sql/plan2/build_insert.go (hunk 3 above)
Comment (ouyuanning) [reply pkg/sql/plan2/build_insert.go:17]: Yes, you are right.  I change the code and add the test case in build_test.go in the last commit。thanks。
and yes some corner not cover，I will fixed it in next commit.

File: pkg/sql/plan2/build_subquery.go (hunk 4 above)
Comment (ouyuanning) [reply pkg/sql/plan2/build_subquery.go:83]: 这个是跟龙冉讨论的结果，对于subquery表达式，其执行的返回值，先看做一个array。
（当时讨论的前提是，把Exists/Any/All 作为母查询的一个Function，那么这几个Function的输入只有一个参数，也就是subquery的执行结果，那么感觉array作为这个参数的类型会比较合适）

Comment (fengttt) [reply pkg/sql/plan2/build_subquery.go:83]: Do we have a tuple or record type?    To me, the return type of subquery is a tuple, that is, (int, string, int, float).   Array is [int, int, ...]

Comment (ouyuanning) [reply pkg/sql/plan2/build_subquery.go]: Get it, I  change the subquery return type to tuple in last commit

**Review by yingfeng — APPROVED** [review]

**Pull request merged**